packages/manager/modules/hub/src/components/billing-summary/controller.js

```diff
@@ -0,0 +1,72 @@
+import get from 'lodash/get';
+
+export default class ManagerHubBillingSummaryCtrl {
+  /* @ngInject */
+  constructor($q, $translate, OvhHttp) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.OvhHttp = OvhHttp;
+  }
+
+  $onInit() {
+    const loadLanguage = this.$q
+      .when(this.me ? this.me.language : this.fetchLanguage())
+      .then((language) => {
+        this.language = language;
+      });
+    const loadBills = this.$q
+      .when(this.bills ? this.bills : this.fetchBills())
```

Comment:
I guess we can make it simple by deciding whether `bills` (and other similar data) should be made mandatory or should it be fetched inside the component itself :)
What do you think?

Reply:
i like the idea that you can either provide or not the data to the component but you are right that the component would be simpler if we choose for example to provides data and the component would just act as a template

i would not bother overthinking it because it's still an angularjs component so it's not something that is meant to be reused in the future

if it's ok for you i'll let it like that otherwise please share your thoughts :)

Reply:
I am ok with it @frenautvh. Sounds good :)